detectable/src/main/java/com/synopsys/integration/detectable/detectables/sbt/dot/SbtCommandArgumentGenerator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * detectable
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.detectable.detectables.sbt.dot;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.detectable.detectables.sbt.parse.SbtResolutionCacheOptions;
+
+public class SbtCommandArgumentGenerator {
+    @Nullable
+    private final List<String> sbtCommandAdditionalArguments;
```

Comment:
I'm not in love with having stateful objects and would rather the additional arguments were passed through. I think Options should be restricted to Extractor/Detectables but I'm sure that is a universally followed rule. I say that because I would imagine its easier to write a unit test for this class if you only need to supply him the arguments list (value he needs) and not the entire Options object.